test/DocumentFormat.OpenXml.Tests/ofapiTest/OpenXmlReaderTest.cs

```diff
@@ -425,6 +425,66 @@ public void PartReaderIgnoreWhitespaceTest(bool ignoreWhitespace)
             reader.Close();
         }
 
+        /// <summary>
+        /// Test that the OpenXmlReader can handle formatted xml.
+        ///</summary>
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void TestIgnoreWhitespaceWithFormattedXml(bool ignoreWhitespace) {
+            const string PartText = @"
+    <w:document xmlns:v=""urn:schemas-microsoft-com:vml"" xmlns:w=""http://schemas.openxmlformats.org/wordprocessingml/2006/main"">
+      <w:body>
+        <w:p>
+          <w:r>
+            <w:t>First Text</w:t>
+          </w:r>
+          <w:r>
+            <w:t>Second Text</w:t>
+          </w:r>
+        </w:p>
+      </w:body>
+    </w:document>";
+
+            UTF8Encoding utf8Encoding = new UTF8Encoding();
+            var exception = Record.Exception(() => {
+                using var stream = new MemoryStream(utf8Encoding.GetBytes(PartText), false);
+                using var reader = OpenXmlReader.Create(stream, false, ignoreWhitespace);
+                while (reader.Read()) {
+                }
+
+                reader.Close();
+            });
+            Assert.Null(exception);
```

Comment:
Instead of asserting null, I'd rather see checks that the reader is output things we'd expect. ie after each `Assert.True(reader.Read())` add an assert for the current state

Reply:
Updated the test cases and added one test case that does not depend on the whitespace setting